### PR TITLE
Fix using Docker plugin with multi-arch manifests

### PR DESCRIPTION
**Pull request by nanonyme**

### CODE REVIEW

**Review by juergbi — APPROVED** [review]

**Comment by nanonyme** [comment]
I messed up and this is missing two fixes that are also blockers.

**Comment by nanonyme** [comment]
Let's please merge this and tag a release asap. We are going to be depending on this in freedesktop-sdk.

File: src/buildstream_plugins/sources/docker.py [hunk 1]
```diff
@@ -329,6 +328,10 @@ class DockerSource(Source):
 
     BST_MIN_VERSION = "2.0"
 
+    # url is deprecated, but accept it as a valid key so that we can raise
+    # a nicer warning.
+    CONFIG_KEYS = ["architecture", "registry-url", "image", "os", "ref", "track", "url"] + Source.COMMON_CONFIG_KEYS
```
Comment (abderrahim) [inline src/buildstream_plugins/sources/docker.py:333]: So I see you added "architecture" here: does this affect the cache keys, or does it only affect tracking?

BTW, moving this here departs from the pattern used by other plugins in this repo.

Comment (abderrahim) [reply src/buildstream_plugins/sources/docker.py:333]: Oh, it's "architecture" and "os". But the same applies.

Comment (nanonyme) [reply src/buildstream_plugins/sources/docker.py:333]: It's actually *old* parameter. This plugin has been incorrectly not allowing it to be used.

Comment (nanonyme) [reply src/buildstream_plugins/sources/docker.py:333]: It does not affect cache key. It affects tracking.

Comment (nanonyme) [reply src/buildstream_plugins/sources/docker.py:333]: These parameters have existed way back in bst_external. This config keys bug existed there too. This plugin has been ported bug for bug into this repo.

Comment (nanonyme) [reply src/buildstream_plugins/sources/docker.py]: Fixed.

**Review by abderrahim — APPROVED** [review]

**Pull request merged**